### PR TITLE
Prevent downgrade of vfs folder

### PR DESCRIPTION
**Pull request by TheOneRing**
image

### CODE REVIEW

File: src/gui/folder.cpp [hunk 1]
```diff
@@ -1299,9 +1310,9 @@ void FolderDefinition::save(QSettings &settings, const FolderDefinition &folder)
 
     // Ensure new vfs modes won't be attempted by older clients
     if (folder.virtualFilesMode == Vfs::WindowsCfApi) {
-        settings.setValue(QLatin1String(versionC), 3);
+        settings.setValue(versionC(), 4);
```
Comment (erikjv) [inline src/gui/folder.cpp:1313]: shouldn't this call `maxSettingsVersion()`?

File: src/gui/folderman.cpp [hunk 2]
```diff
@@ -37,8 +37,20 @@
 #include <QNetworkProxy>
 
 namespace {
-const char versionC[] = "version";
-const int maxFoldersVersion = 1;
+/*
+ * [Accounts]
+ * 0\version=1
+ */
+auto versionC()
```
Comment (erikjv) [inline src/gui/folderman.cpp:44]: Isn't this a duplicate of the one in `folder.cpp`?

File: src/gui/folderman.cpp [hunk 3]
```diff
@@ -403,12 +415,12 @@ void FolderMan::backwardMigrationSettingsKeys(QStringList *deleteKeys, QStringLi
...
     auto processSubgroup = [&](const QString &name) {
         settings->beginGroup(name);
-        const int foldersVersion = settings->value(QLatin1String(versionC), 1).toInt();
+        const int foldersVersion = settings->value(versionC(), 1).toInt();
         if (foldersVersion <= maxFoldersVersion) {
             const auto &childGroups = settings->childGroups();
             for (const auto &folderAlias : childGroups) {
                 settings->beginGroup(folderAlias);
-                const int folderVersion = settings->value(QLatin1String(versionC), 1).toInt();
+                const int folderVersion = settings->value(versionC(), 1).toInt();
```
Comment (erikjv) [inline src/gui/folderman.cpp:423]: `1` is another magic number. Can you make constants somewhere in 1 place (in a header file?) and add some comments to them? I've now seen 1, 2, and 4. Oh, and there was a 3, but that's now "old"?

**Review by erikjv — CHANGES_REQUESTED** [review]

File: src/gui/folderman.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/gui/folderman.cpp:44]: Both are version but in different section that are evaluated independently.
But yes...
However refactoring the settings stuff is stuff for a major release...

File: src/gui/folderman.cpp (hunk 3 above)
Comment (TheOneRing) [reply src/gui/folderman.cpp:423]: For some reason someone decided to start to count with 1, that's why we default to 1 if no version at all is set.

File: src/gui/folder.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/folder.cpp:1313]: Hmm as you can see two lines later it doesn't always need to be the max version.
But I'll extract the two versions.

**Review by erikjv — APPROVED** [review]

**Pull request merged**